mesonbuild/optinterpreter.py

```diff
@@ -200,11 +201,8 @@ def evaluate_statement(self, node):
             raise OptionException('Only calls to option() are allowed in option files.')
         (posargs, kwargs) = self.reduce_arguments(node.args)
 
-        # FIXME: Cannot use FeatureNew while parsing options because we parse
-        # it before reading options in project(). See func_project() in
-        # interpreter.py
-        #if 'yield' in kwargs:
-        #    FeatureNew('option yield', '0.45.0').use(self.subproject)
```

Comment:
There should be an assertion here that `project_meson_versions` is set, since assumptions about the initialization order break easily during a refactoring (especially if there are no comments).

Reply:
I've done the best I can to address this in the latest version of the series. There is a corner case that's in our test suite that makes it kinda hard to solve generally, so the approach I have is a bit of a hack (I really don't want to rewrite the way this detection works completely), that corner is if a FeatureNew/FeatureDeprecated is triggered in an argument to project(), such as
```meson
project('foo', version : run_command(..., check : true))
```
So I've changed the type of project_meson_versions to a defaultdict so there is always a value, and added a comment to project() implementation

Reply:
Ah, I didn't see this corner case immediately...